docs/user-guide/gaps.md

```diff
@@ -180,4 +180,138 @@ is sufficient both by running the basic sanity rule,
 and if loops appear only under certain conditions, 
 to write specialized sanity rules that force the Prover to reason about these 
 particular code paths.
-Mutation testing can also be useful here.
\ No newline at end of file
+Mutation testing can also be useful here.
+
+### 'Hidden' compiler-generated loops
+
+#### Copying Solidity memory arrays to storage
+
+When the Solidity compiler generates code for copying 
+a non-primitive object (could be a `bytes` buffer or a `struct` with a `bytes` field),
+it generates two loops.
+The first loop resets any previous remaining data written into the target storage slot.
+The second loop copies the new object into the relevant storage slot.
+
+Consider the following contract:
+```solidity
+// MemoryToStorage.sol
+contract MemoryToStorage {
+  struct ScheduledExecution {
+    address where;
+    bool execute;
+    bytes data;
+  }
+
+  ScheduledExecution[] myArray;
+
+  function testPush(address where, bool executed) public {
+    bytes memory data = abi.encodeWithSelector(
+      this.testPush.selector,
+      "aa"
+    );
+    myArray.push(ScheduledExecution(where, executed, data));
+  }
+}
+```
+
+The push to `myArray` generates the two aforementioned loops. 
+If we wish to analyze this code with the Prover, there are two questions to be answered:
+
+1. Do we need to set a value for `--loop_iter` which is bigger than 1?
```

Comment:
Isn't the prover smart enough to unroll the loop enough times for this case? The string is static